packages/sources/view-function/src/transport/function.ts

```diff
@@ -0,0 +1,70 @@
+import { Transport, TransportDependencies } from '@chainlink/external-adapter-framework/transports'
+import { ResponseCache } from '@chainlink/external-adapter-framework/cache/response'
+import { Requester } from '@chainlink/external-adapter-framework/util/requester'
+import { AdapterRequest, AdapterResponse } from '@chainlink/external-adapter-framework/util'
+import { BaseEndpointTypes, inputParameters } from '../endpoint/function'
+import { ethers, utils } from 'ethers'
+
+export type FunctionTransportTypes = BaseEndpointTypes
+
+export class FunctionTransport implements Transport<FunctionTransportTypes> {
+  name!: string
+  responseCache!: ResponseCache<FunctionTransportTypes>
+  requester!: Requester
+  provider!: ethers.providers.JsonRpcProvider
+
+  async initialize(
+    dependencies: TransportDependencies<FunctionTransportTypes>,
+    _adapterSettings: FunctionTransportTypes['Settings'],
+    _endpointName: string,
+    transportName: string,
+  ): Promise<void> {
+    this.responseCache = dependencies.responseCache
+    this.requester = dependencies.requester
+    this.name = transportName
+    this.provider = new ethers.providers.JsonRpcProvider(
+      _adapterSettings.ETHEREUM_RPC_URL,
+      _adapterSettings.ETHEREUM_CHAIN_ID,
+    )
+  }
```

Comment:
We can avoid having to initialize a couple of these. Also I don't think we're using a requester on this adapter right?
```suggestion
  provider!: ethers.providers.JsonRpcProvider

  async initialize(
    dependencies: TransportDependencies<FunctionTransportTypes>,
    adapterSettings: FunctionTransportTypes['Settings'],
    endpointName: string,
    transportName: string,
  ): Promise<void> {
    await super.initialize(dependencies, adapterSettings, endpointName, transportName)
    this.provider = new ethers.providers.JsonRpcProvider(
      adapterSettings.ETHEREUM_RPC_URL,
      adapterSettings.ETHEREUM_CHAIN_ID,
    )
  }
```